src/compiler/checker.ts

```diff
@@ -2772,10 +2772,12 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         return combined;
     }
 
-    function mergeSymbolTable(target: SymbolTable, source: SymbolTable, unidirectional = false) {
+    function mergeSymbolTable(target: SymbolTable, source: SymbolTable, unidirectional = false, parent?: Symbol) {
         source.forEach((sourceSymbol, id) => {
             const targetSymbol = target.get(id);
-            target.set(id, targetSymbol ? mergeSymbol(targetSymbol, sourceSymbol, unidirectional) : getMergedSymbol(sourceSymbol));
+            const merged = targetSymbol ? mergeSymbol(targetSymbol, sourceSymbol, unidirectional) : getMergedSymbol(sourceSymbol);

```

Comment:
I can't remember: can `targetSymbol` or `sourceSymbol` be returned from `mergeSymbol` or `getMergedSymbol`? Or is the result always a fresh transient symbol?

The reason I'm asking is that I want to know what happens if we already have an existing `parent`. Is it guaranteed to be the same as `parent`? Related to it in some way?